src/Warnings.hpp

```diff
@@ -134,6 +134,13 @@ THE SOFTWARE.
   #define RESTORE_WARNINGS \
     __pragma(warning(pop))
 
+#elif defined(__GNUC__)
+
+  #define SUPPRESS_WARNINGS _Pragma("GCC diagnostic push") \
+    _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
+
+  #define RESTORE_WARNINGS _Pragma("GCC diagnostic pop")
+
```

Comment:
Interesting that you've to do something here.  In `$toplevel/CMakeLists.txt` we actually don't set any warning level for GCC.  Maybe that should be added in this branch, too?